src/layers/models/layersManager.ts

```diff
@@ -107,8 +106,20 @@ export class LayersManager {
 
     await this.validateJobNotRunning(productId, productType);
 
+    const isGpkg = this.ingestionValidator.validateIsGpkg(files);
+    if (isGpkg) {
+      this.ingestionValidator.validateGpkgFiles(files, originDirectory);
+      const grids: Grid[] = [];
+      files.forEach((file) => {
+        const sqliteClient = new SQLiteClient(this.config, this.logger, file, originDirectory);
+        const grid: Grid = sqliteClient.getGrid() as Grid;
+        grids.push(grid);
+      });
```

Comment:
refactor to a method of getGpkgGrids(): Grid[]
Note: i prefer it will be in GPKG dedicated file (and reuse it in method validateGpkgGrid) - but i'll accept it if you'le do it in this file

Reply:
done